exchanges/binance/binance.go

```diff
@@ -37,31 +86,2085 @@ const (
+	if limit > 0 && limit < 500 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if startTime.After(endTime) {
+			return resp, errors.New("startTime cannot be after endTime")
+		}
+		params.Set("start_time", strconv.FormatInt(startTime.Unix(), 10))
+		params.Set("end_time", strconv.FormatInt(endTime.Unix(), 10))
+	}
+	return resp, b.SendHTTPRequest(b.API.Endpoints.URL+ufuturesTopPositionsRatio+params.Encode(), limitDefault, &resp)
+}
+
+// UGlobalLongShortRatio gets the global long/short ratio data for ufutures
+func (b *Binance) UGlobalLongShortRatio(symbol, period string, limit int64, startTime, endTime time.Time) ([]ULongShortRatio, error) {
+	var resp []ULongShortRatio
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	if !common.StringDataCompare(uValidPeriods, period) {
+		return resp, fmt.Errorf("invalid period")
+	}
+	params.Set("period", period)
+	if limit > 0 && limit < 500 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if startTime.After(endTime) {
+			return resp, errors.New("startTime cannot be after endTime")
+		}
+		params.Set("start_time", strconv.FormatInt(startTime.Unix(), 10))
+		params.Set("end_time", strconv.FormatInt(endTime.Unix(), 10))
+	}
+	return resp, b.SendHTTPRequest(b.API.Endpoints.URL+ufuturesLongShortRatio+params.Encode(), limitDefault, &resp)
+}
+
+// UTakerBuySellVol gets takers' buy/sell ratio for ufutures
+func (b *Binance) UTakerBuySellVol(symbol, period string, limit int64, startTime, endTime time.Time) ([]UTakerVolumeData, error) {
+	var resp []UTakerVolumeData
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	if !common.StringDataCompare(uValidPeriods, period) {
+		return resp, fmt.Errorf("invalid period")
+	}
+	params.Set("period", period)
+	if limit > 0 && limit < 500 {
+		params.Set("limit", strconv.FormatInt(limit, 10))
+	}
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if startTime.After(endTime) {
+			return resp, errors.New("startTime cannot be after endTime")
+		}
+		params.Set("start_time", strconv.FormatInt(startTime.Unix(), 10))
+		params.Set("end_time", strconv.FormatInt(endTime.Unix(), 10))
+	}
+	return resp, b.SendHTTPRequest(b.API.Endpoints.URL+ufuturesLongShortRatio+params.Encode(), limitDefault, &resp)
+}
+
+// UFuturesNewOrder sends a new order for ufutures
+func (b *Binance) UFuturesNewOrder(symbol, side, positionSide, orderType, timeInForce,
+	reduceOnly, newClientOrderID, closePosition, workingType, newOrderRespType string,
+	quantity, price, stopPrice, activationPrice, callbackRate float64) (UOrderData, error) {
+	var resp UOrderData
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("side", side)
+	if positionSide != "" {
+		if !common.StringDataCompare(validPositionSide, positionSide) {
+			return resp, fmt.Errorf("invalid positionSide")
+		}
+		params.Set("positionSide", positionSide)
+	}
+	params.Set("type", orderType)
```

Comment:
nice spotting. shame ur not as good at spotting trades

Reply:
![image](https://user-images.githubusercontent.com/9391715/96948587-c2c1e580-1531-11eb-8b4f-2fd6c951e03b.png)
Thanks, this will help me from that awesome burn.